tests/test_hawkdove.py

```diff
@@ -0,0 +1,263 @@
+import math
+from unittest.mock import Mock, patch
+from collections import Counter
+
+import pytest
+
+from simulatingrisk.hawkdove.model import (
+    HawkDoveAgent,
+    Play,
+    HawkDoveSingleRiskModel,
+    HawkDoveSingleRiskAgent,
+)
+
+
+def test_agent_neighbors():
+    # initialize model with a small grid, neighborhood of 8
+    model = HawkDoveSingleRiskModel(3, play_neighborhood=8, agent_risk_level=4)
+    # every agent should have 8 neighbors when diagonals are included
+    assert all([len(agent.play_neighbors) == 8 for agent in model.schedule.agents])
+
+    # neighborhood of 4
+    model = HawkDoveSingleRiskModel(3, play_neighborhood=4, agent_risk_level=2)
+    assert all([len(agent.play_neighbors) == 4 for agent in model.schedule.agents])
+
+    # neighborhood of 24 (grid needs to be at least 5x5)
+    model = HawkDoveSingleRiskModel(5, play_neighborhood=24, agent_risk_level=5)
```

Comment:
I can run a model  HawkDoveSingleRiskModel(3, play_neighborhood=24, agent_risk_level=3), and this does not trow an error.  There seems to be a check from grid size to playable neighborhood missing? or not working? 
I do not know how the grid neighborhood works for small grids with radius 2 neighbors. Are agents then just repeated?